pkg/clients/elasticache/elasticache.go

```diff
@@ -816,5 +840,40 @@ func IsClusterUpToDate(name string, in *cachev1alpha1.CacheClusterParameters, ob
 	}
 	GenerateCluster(name, *in, desired)
 
+	if desired.EngineVersion != nil {
+		observedVersion := observed.EngineVersion
+		desiredVersion := desired.EngineVersion
+
+		observedVersionSplit := strings.Split(aws.ToString(observedVersion), ".")
+		desiredVersionSplit := strings.Split(aws.ToString(desiredVersion), ".")
+		if observedVersionSplit[0] != desiredVersionSplit[0] {
+			return false, nil
+		}
+		if len(desiredVersionSplit) > 1 {
+			if observedVersionSplit[1] != desiredVersionSplit[1] {
+				return false, nil
+			}
+		}
+		// to ignore in following equal
+		desired.EngineVersion = observed.EngineVersion
+	}
+
 	return cmp.Equal(desired, observed, cmpopts.EquateEmpty(), cmpopts.IgnoreTypes(document.NoSerde{})), nil
 }
+
+func getVersion(version *string) (*string, error) {
```

Comment:
If we want to be sure we are sending a request with a supported version, we could call `DescribeCacheEngineVersions` and store them in a map, then check against it. Might be an overkill though.

Reply:
think this is overkill :D